src/coreComponents/python/modules/pygeosx_tools_package/pygeosx_tools/well_log.py

```diff
@@ -0,0 +1,85 @@
+
+import numpy as np
+
+
+def parse_las(fname, variable_start='~CURVE INFORMATION', body_start='~A'):
+    """
+    Parse an las format log file
+
+    Args:
+        fname (str): Path to the log file
+        variable_start (str): A string that indicates the start of variable header information (default = '~CURVE INFORMATION')
+        body_start (str): a string that indicates the start of the log body (default = '~A')
+
+    Returns:
+        np.ndarray: a dict containing the values and unit definitions for each variable in the log
+    """
+    variable_names = []
+    variable_units = []
+    variable_values = []
+
+    with open(fname) as f:
+        file_location = 0
+        for line in f:
+            # Preamble
+            if (file_location == 0):
+                if variable_start in line:
+                    file_location += 1
+
+            # Variable definitions
+            elif (file_location == 1):
+                if ('#' not in line):
+                    if body_start in line:
+                        file_location += 1
+                    else:
+                        tmp = line.split()
+                        variable_names.append(tmp[0])
+                        variable_units.append(tmp[1])
+
+            # Body
+            else:
+                variable_values.append([float(x) for x in line.split()])
+
+    # Format results
+    results = {}
+    variable_values = np.array(variable_values)
+
+    for ii in range(0, len(variable_names)):
```

Comment:
Use enumerate instead.

Or could it be possible to use `zip` (I do not know with `numpy.squeeze`)
```python
for n, u, v in zip(variable_names, variable_units, variable_values):
        results[n] = {'units': u, 'values': np.squeeze(?v?)}
```